lab_1_keywords_tfidf/main.py

```diff
@@ -4,106 +4,144 @@
 """
 from typing import Optional, Union
 
+import math
+import string
+
+
+def dict_type_check(dictionary: dict, key_type: Union[type, tuple], val_type: Union[type, tuple]) -> bool:
+    """
+    Checks whether the object is a dictionary and its keys and values are of the expected type
+    Parameters:
+    dictionary:
+    key_type: the expected key type
+    val_type: the expected value type
+    Returns:
+    bool: True if dictionary keys and values are of the expected type, False otherwise
+    """
+    if (isinstance(dictionary, dict)
+            and all(isinstance(key, key_type) for key in dictionary.keys())
+            and all(isinstance(value, val_type) for value in dictionary.values())):
+        return True
+    return False
+
 
 def clean_and_tokenize(text: str) -> Optional[list[str]]:
     """
     Removes punctuation, casts to lowercase, splits into tokens
-
     Parameters:
     text (str): Original text
-
     Returns:
     list[str]: A sequence of lowercase tokens with no punctuation
-
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if isinstance(text, str):
+        for punc in string.punctuation:
+            text = text.replace(punc, '')
+        clean_words = text.lower().split()
+        return clean_words
+    return None
```

Comment:
consider using early return
```
if not <data is valid>:
   return None
<main logic for the function>
```
This way code is more readable

Reply:
it concerns all the functions

Reply:
done